aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,273 @@
+  :linenos:
+  :dedent: 8
+  :language: c#
+  :lines: 21-34
+  :emphasize-lines: 7-8,10-13
+  
+You can see on line 13 how ``ApplicationServices`` is used to access the ``IInlineConstraingResolver`` dependency; if this were done from a ``RouteBuilder`` extension method, access to the services would be done through the ``ServiceProvider`` property.
+
+Once you've finished adding routes to the ``RouteBuilder`` instance, call ``UseRouter`` and pass it the result of the ``RouteBuilder.Build`` method.
+
+.. tip:: If you are only configuring a single route, you can simply call ``app.UseRouter`` and pass in the ``IRouter`` instance you wish to use, bypassing the need to use a ``RouteBuilder``.
+
+The sample shown above uses the default empty web template, which includes the "Hello World" request delegate shown on lines 16-19. The route added on line 8 will only match requests of the form "hello/{name}" where `name` is constrained to be alphabetical. Requests that match this will be handled by a custom ``IRouter`` implementation, ``HelloRouter``.
+
+.. literalinclude:: routing/sample/RoutingSample/HelloRouter.cs
+  :linenos:
+  :language: c#
+
+``HelloRouter`` checks to see if ``RouteData`` includes a value for the key ``name``. If not, it immediately returns without handling the request. Otherwise, the request is handled (by writing out "Hi {name}!") and the ``RouteContext`` is updated to note that the request was handled. This prevents additional routes from handling the request.
+
+This route was configured to use an inline constraint (see below), signified by the ":alpha" in the route template for the name route value. This constraint limits which requests this route will handle, in this case to alphabetical values for ``name``. Thus, a request for "/hello/steve" will be handled, but a request to "/hello/123" will not (instead, the request will pass through to the "Hello World!" request delegate).
+
+Template Routes
+---------------
+The most common way to define routes is using ``TemplateRoute`` and route template strings. In a typical MVC app, you might use a default template route with a string like this one: 
+
+.. image:: /fundamentals/routing/_static/default-mvc-routetemplate.png
+
+Tokens within curly braces (``{ }``) define `route value` placeholders which will be bound if the route is matched. You can define more than one route value placeholder in a route segment, but they must be separated by a literal value. For example "/{controller}{action" would not be a valid route, since there is no literal value between ``{controller}`` and ``{action}``. These route value placeholders must have a name, and may have additional attributes specified.
+
+The following table demonstrates the available route value placeholder options and `route value constraints <https://github.com/aspnet/Routing/blob/1.0.0-rc1/src/Microsoft.AspNet.Routing/RouteOptions.cs>`_.
+
+.. list-table:: Route Template Values
```

Comment:
I was shown a trick by the RST authors - they said you should only create tables in their sandbox - once it works then transfer to you file. I'm trying to get the URL of their test sandbox. If it's not too much work, it would be better to have a table that didn't require scrolling.